modules/azure-landing-zone/rg.tf

```
@@ -0,0 +1,6 @@
+resource "azurerm_resource_group" "main" {
```

this probably doesn't belong here? should be provisioned elsewhere?
```suggestion
data "azurerm_resource_group" "main" {
```

---

> this probably doesn't belong here? should be provisioned elsewhere?

nope. I dsagree. The landing zone should be kept entirely separate to all the other resources. Hence we should keep all the resources that constitutes the landing zone to be in one resource group. However this resource group should not be used by any other resources. Hence the name of the resource group will not be available as an output.

---

what if the service principal doesn't have permissions to create the resource group though?
we've been moving away from subscription level permissions to resource group level permissions to limit the blast radius from mistakes, and reduce 'god' level permissions